fix(autonomi): resolve libinstant error

b-zee opened the pull request:
See
https://github.com/rustwasm/wasm-bindgen/discussions/3500#discussioncomment-6334669
for an example of errors that happened when running the PUT test.

oetyng commented on sn_client/examples/churn.rs:
```diff
@@ -69,112 +63,33 @@ async fn main() -> Result<()> {
+        .build()?;
+    testnet.launch_genesis(None, vec![])?;
+    testnet.launch_nodes(node_count, &network_contacts_path, vec![])?;
+    testnet.configure_network_contacts(&network_contacts_path)?;
```
So nice simplification of setup here.

oetyng commented on sn_client/examples/churn.rs:
```diff
@@ -69,112 +63,33 @@ async fn main() -> Result<()> {
-        .map_err(|error| eyre!(error))
-        .and_then(|launch| launch.run())
-        .wrap_err("Error adding nodes to the testnet")?;
+    testnet.launch_nodes(ADDITIONAL_NODE_COUNT, &network_contacts_path, vec![])?;
```
And here. ❤

oetyng commented on testnet/src/lib.rs:
```diff
@@ -0,0 +1,963 @@
+    use color_eyre::Result;
+    use mockall::predicate::*;
+
+    const NODE_LAUNCH_INTERVAL: u64 = 0;
```
I wonder if some of the strings that are frequently used in the tests should be local constants?

Like `"local-test-network"`, `"sn-node-genesis"`, `"safenode"`, etc.

oetyng commented on testnet/src/lib.rs:
```diff
@@ -0,0 +1,963 @@
+            if i < end {
+                info!(
+                    "Delaying for {} seconds before launching the next node",
+                    self.node_launch_interval / 1000
```
This is old stuff, but I've seen the interval being modified with a number here and there. It seems we for some reason want something else than node launch interval.

I wonder if those places could get a named variable with some docs, so we know why suddenly some other interval is used here or there?

oetyng commented on testnet/src/main.rs:
```diff
@@ -0,0 +1,281 @@
+
+use clap::Parser;
+use color_eyre::{eyre::eyre, Help, Result};
+use std::path::PathBuf;
```
Can be collapsed into ln 33.

oetyng commented on .github/workflows/merge.yml:
```diff
@@ -242,15 +242,11 @@ jobs:
         id: section-startup
         env:
-          RUST_LOG: "sn_node,sn_comms,sn_consensus,sn_fault_detection,sn_interface=trace"
+          RUST_LOG: "safenode,sn_node,sn_comms,sn_consensus,sn_fault_detection,sn_interface=trace"
```
Is the intention perhaps to replace `sn_node` in these declarations? I see it is replaced in some places further down.

jacderida replied:
As mentioned on Slack, this was intentional.

oetyng reviewed (approved):
Great work! On many levels.
I really like how you write the docs, written as they should IMO. Thoroughly, describing what it is, how it is used and some motivations on why this or that choice has been made/not made.

b-zee committed:
fix(autonomi): resolve libinstant error

See
https://github.com/rustwasm/wasm-bindgen/discussions/3500#discussioncomment-6334669
for an example of errors that happened when running the PUT test.
